broker-core/src/main/java/io/zeebe/broker/workflow/processor/CatchEventBehavior.java

```diff
@@ -78,20 +92,15 @@ public void subscribeToEvents(
             event.getTimer(),
             context.getOutput().getStreamWriter());
       } else if (event.isMessage()) {
-        subscribeToMessageEvent(context, event);
+        subscribeToMessageEvent(context, event, extractedCorrelationKeys.get(event.getId()));
       }
     }
 
     context
         .getStateDb()
         .getEventScopeInstanceState()
         .createInstance(context.getRecord().getKey(), supplier.getInterruptingElementIds());
-  }
-
-  private void validateEventSubscription(BpmnStepContext<?> context, ExecutableCatchEvent event) {
-    if (event.isMessage()) {
-      extractCorrelationKey(context, event.getMessage());
-    }
+    extractedCorrelationKeys.clear();
```

Comment:
The keys are being cleared on [line 253](https://github.com/zeebe-io/zeebe/pull/1987/files#diff-4aefa6833887bfabe091652ce4ba0b8cR253)

Reply:
Sorry, this was supposed to be part of the review but I clicked the wrong button

Reply:
The idea was that we don't need to keep the memory around until the next time this is called, but that's premature optimization so I'll remove it. The one at 253 is the important one.